### PR TITLE
MI300 documentation

### PR DESCRIPTION
**Pull request by neon60**
Mi300 documentation

### CODE REVIEW

File: docs/conceptual/gpu-arch.md [hunk 1]
```diff
@@ -7,9 +7,20 @@
 
 # GPU architecture documentation
 
-:::::{grid} 1 1 2 2
+:::::{grid} 1 1 3 2
```
Comment (LisaDelaney) [inline docs/conceptual/gpu-arch.md:10]:
```suggestion
:::::{grid} 1 1 2 2
```

**Review by saadrahim — APPROVED** [review]

File: docs/conceptual/gpu-arch/mi300-performance-counters.rst [hunk 2]
```diff
@@ -0,0 +1,668 @@
...
+  "``SQ_LDS_IDX_ACTIVE``", "Cycles", "Number of cycles LDS is used for indexed operations"
+
+Miscellaneous counters
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. csv-table::
+  :widths: 30, 10, 60
+  :header: "Hardware counter", "Unit", "Definition"
+
+  "``SQ_IFETCH``", "Count", "Number of instruction fetch requests from `L1I` cache, in 32-byte width"
```
Comment (LisaDelaney) [inline docs/conceptual/gpu-arch/mi300-performance-counters.rst:303]: What is L1I?

Comment (Rmalavally) [reply docs/conceptual/gpu-arch/mi300-performance-counters.rst:303]: Level 1 (L1) Cache; however, an SME can confirm. Also, the 'i ' must be in lowercase - L1i. Level 1 instructions.

File: docs/conceptual/gpu-arch/mi300-performance-counters.rst [hunk 3]
```diff
@@ -0,0 +1,668 @@
...
+
+.. csv-table::
+  :widths: 30, 5, 60, 5
+  :header: "Hardware counter", "Unit", "Definition", "Value range for ``n``"
+
+  "``TCP_GATE_EN1[n]``", "Cycles", "Number of cycles vector L1 data cache interface clocks are turned on", "0-15"
+  "``TCP_GATE_EN2[n]``", "Cycles", "Number of cycles vector L1 data cache core clocks are turned on", "0-15"
+  "``TCP_TD_TCP_STALL_CYCLES[n]``", "Cycles", "Number of cycles texture data unit stalls vector L1 data cache", "0-15"
+  "``TCP_TCR_TCP_STALL_CYCLES[n]``", "Cycles", "Number of cycles texture cache router stalls vector L1 data cache", "0-15"
+  "``TCP_READ_TAGCONFLICT_STALL_CYCLES[n]``", "Cycles", "Number of cycles tagram conflict stalls on a read", "0-15"
```
Comment (LisaDelaney) [inline docs/conceptual/gpu-arch/mi300-performance-counters.rst:395]: what is tagram?

Comment (Rmalavally) [reply docs/conceptual/gpu-arch/mi300-performance-counters.rst:395]: Probably Cache tag RAM. We have used this in our documentation before. Better documented as tag_RAM or tag_ram. Please confirm with the SME.

File: docs/conceptual/gpu-arch/mi300-performance-counters.rst [hunk 4]
```diff
@@ -0,0 +1,668 @@
...
+  ``TCC_TAG_STALL[n]``, probes can stall the pipeline at a variety of places. There is no single point that
+  can accurately measure the total stalls
+
+MI300 derived metrics list
+==============================================================
+
+.. csv-table::
+  :widths: 30, 70
+  :header: "Hardware counter", "Definition"
+
```
Comment (LisaDelaney) [inline docs/conceptual/gpu-arch/mi300-performance-counters.rst:532]: The entries in this table don't seem to be in a specific order. I would suggest putting them in a logical order...and maybe breaking the table into smaller tables (if there are logical clusters)

File: docs/conceptual/gpu-arch/mi300-performance-counters.rst (hunk 2 above)
Comment (neon60) [reply docs/conceptual/gpu-arch/mi300-performance-counters.rst]: I copied over from here.

https://raw.githubusercontent.com/ROCm/ROCm/develop/docs/conceptual/gpu-arch/mi200-performance-counters.md

I am fixing at both location, I also think we need lowercase.

File: docs/conceptual/gpu-arch/mi300-performance-counters.rst (hunk 3 above)
Comment (neon60) [reply docs/conceptual/gpu-arch/mi300-performance-counters.rst:395]: Have to effect here too:

https://raw.githubusercontent.com/ROCm/ROCm/develop/docs/conceptual/gpu-arch/mi200-performance-counters.md

File: docs/conceptual/gpu-arch/mi300-performance-counters.rst (hunk 4 above)
Comment (neon60) [reply docs/conceptual/gpu-arch/mi300-performance-counters.rst]: Made the split up and more.

File: docs/conceptual/gpu-arch/mi300-mi200-performance-counters.rst [hunk 5]
```diff
@@ -613,38 +613,45 @@ MI300 and Mi200 series derived metrics list
   "``WriteSize``", "Total kilobytes written to the video memory. This is measured with all extra fetches and any cache or memory effects taken into account."
   "``WriteUnitStalled``", "Percentage of GPU time the write unit is stalled. Value range: 0% (optimal) to 100%."
 
-Hardware counters by/over all texture addressing unit instances
------------------------------------------------------------------
+Hardware counters by and over all texture addressing unit instances
```
Comment (LisaDelaney) [inline docs/conceptual/gpu-arch/mi300-mi200-performance-counters.rst:616]: @neon60 thoughts?

Comment (neon60) [reply docs/conceptual/gpu-arch/mi300-mi200-performance-counters.rst:616]: by or over?

Comment (neon60) [reply docs/conceptual/gpu-arch/mi300-mi200-performance-counters.rst:616]: Your changes look good.

File: docs/conceptual/gpu-arch/mi300-mi200-performance-counters.rst [hunk 6]
```diff
@@ -0,0 +1,759 @@
...
+  "``TCP_TOTAL_ACCESSES_sum``", "Total number of vector L1d accesses"
+  "``TCP_TOTAL_READ_sum``", "Total number of vector L1d read accesses"
+  "``TCP_TOTAL_WRITE_sum``", "Total number of vector L1d write accesses"
+  "``TCP_TOTAL_ATOMIC_WITH_RET_sum``", "Total number of vector L1d atomic requests with return"
+  "``TCP_TOTAL_ATOMIC_WITHOUT_RET_sum``", "Total number of vector L1d atomic requests without return"
+  "``TCP_TOTAL_WRITEBACK_INVALIDATES_sum``", "Total number of vector L1d writebacks and invalidates"
+  "``TCP_VOLATILE_sum``", "Total number of L1 volatile pixels or buffers from texture addressing unit"
+  "``TCP_WRITE_TAGCONFLICT_STALL_CYCLES_sum``", "Total number of cycles tagram conflict stalls on a write"
+
+Hardware counter over all texture data unit instances
```
Comment (LisaDelaney) [inline docs/conceptual/gpu-arch/mi300-mi200-performance-counters.rst:747]: @neon60 can you verify the accuracy of this heading?

File: docs/conceptual/gpu-arch/mi300-performance-counters.rst (hunk 3 above)
Comment (neon60) [reply docs/conceptual/gpu-arch/mi300-performance-counters.rst]: Merged the two files. @LisaDelaney can you ask SME?

File: docs/conceptual/gpu-arch/mi300-mi200-performance-counters.rst (hunk 6 above)
Comment (neon60) [reply docs/conceptual/gpu-arch/mi300-mi200-performance-counters.rst:747]: Yeah, It looks much better.

**Comment by neon60** [comment]
@ooreilly @eemorsi please review this PR, we would like to merge this in.

**Comment by neon60** [comment]
@LisaDelaney Rebased our changes, please approve the PR, id it looks good.

**Review by MathiasMagnus — DISMISSED** [review]

**Review by LisaDelaney — DISMISSED** [review]

**Comment by neon60** [comment]
Rebased the branch.

**Review by saadrahim — APPROVED** [review]

**Pull request merged**